pkg/jwt/jwt.go

```diff
@@ -0,0 +1,85 @@
+package jwt
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/marstr/guid"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/host"
+)
+
+type JWT struct {
+	options JWTOptions
+}
+
+func New(ctx context.Context, hn host.SonrHost) JWT {
+	options = JWTOptions{}
+	options.secret = []byte(hn.Config().JWT_Secret)
+	options.singingMethod = hn.Config().JWT_SigningMethod
+	options.ttl = hn.Config().JWT_Expiration
+
+	return JWT{
+		options: options,
+	}
+}
+
+func DefaultNew() JWT {
+	return JWT{
+		options: options.DefaultTestConfig(),
+	}
+}
+
+/*
+
+ */
+func (j *JWT) Generate(doc did.Document) (string, error) {
+	if doc == nil {
```

Comment:
Since you're not using a pointer argument, this will never be true (only pointers can be nil). Consider instead checking that the fields in the struct are valid.